### DISCUSSION TITLE
EditOnClick Error

### DISCUSSION THREAD

**Discussion by musumbidenis** [Power-Components/livewire-powergrid#665] [Q&A]
I'm  getting the following error when I do an editOnClick function on a row:
**_Property [$first_name] not found on component: [users]_**
Anyone with a clue to a possible fix, i'll highly appreciate

**Comment by luanfreitasdev** [comment]
Hello,

When you add editOnClick in a column it expects the field name to be a property in the component

```php
public ?string $name = null;

protected array $rules = [
     'name.*' => ['required', 'min:6'],
];
```
https://livewire-powergrid.com/#/table/cell-actions-buttons?id=validation

**Reply by musumbidenis** [reply]
Thank you.
I corrected that and bumped into the following error;
**_Cannot auto-initialize an array inside property App\Http\Livewire\Users::$first_name of type ?string_**

**Reply by luanfreitasdev** [reply]
Hello. Excuse me. The correct type is array

**Reply by musumbidenis** [reply]
Hello, made the correction and still got this error;
**_Call to a member function update() on null_**
Where might I have gone wrong?

**Reply by musumbidenis** [reply]
Hello @luanfreitasdev kindly assist

**Reply by luanfreitasdev** [reply]
Hello. please check for changes between versions. V3 does not support update method

https://livewire-powergrid.com/#/get-started/upgrade-guide?id=deprecations

**Reply by mariano-pualiu** [reply]
@luanfreitasdev can you explain **why** is it expecting an array specifically for the property?
